docs/core/tools/create-custom-templates.md

```diff
@@ -0,0 +1,232 @@
+- You can create a template from an existing project quickly by merely adding a *template.json* configuration file to the project.
+
+Files and folders stored in the template aren't limited to formal .NET project types, such as a .NET Core or .NET Framework solutions. Source files and folders may consist of any content that you wish to create when the template is used, even if the template engine produces just one file for its output, such as a configuration file or a solution file. For example, you can create a template that contains a single *web.config* source file that creates a modified *web.config* file. The modifications to source files are based on logic and settings you've provided in the *template.json* configuration file along with values provided by the user passed as options to the `dotnet new <TEMPLATE>` command.
+
+### template.json
+
+The *template.json* file is placed in a *.template.config* folder in the root directory of the template. The file provides configuration information to the template engine. The minimum configuration requires the members shown in the following table, which is sufficient to add to a .NET Core app to produce a functional template.
+
+| Member            | Type          | Description |
+| ----------------- | ------------- | ----------- |
+| `author`          | string        | The author of the template. |
+| `classifications` | array(string) | Zero or more characteristics of the template that a user might search for it by. |
+| `identity`        | string        | A unique name for this template. |
+| `name`            | string        | The name for the template that users should see. |
+| `shortName`       | string        | A default shorthand for selecting the template (applies to environments where the template name is specified by the user - not selected via a GUI). |
+
+**Example**
+
+```json
+{
+  "author": "Catalina Garcia",
+  "classifications": [ "Common", "Console" ],
+  "identity": "GarciaSoftware.ConsoleTemplate.CSharp",
+  "name": "Garcia Software Console Application",
+  "shortName": "garciaconsole"
+}
+```
+
+The full schema for the *template.json* file is found at the [JSON Schema Store](http://json.schemastore.org/template).
+
+## .NET default templates
+
+When you install the [.NET Core SDK](https://www.microsoft.com/net/download/core), you receive over a dozen built-in templates for creating projects and files, including console apps, class libraries, unit test projects, ASP.NET Core apps (including [Angular](https://angular.io/) and [React](https://facebook.github.io/react/) projects), and configuration files. To list the built-in templates, execute the `dotnet new` command with the `-l|--list` option:
+
+```console
+dotnet new -l
+```
+
+## Packing a template into a NuGet package
+
+Currently, a custom template is packed with [nuget.exe](https://dist.nuget.org/win-x86-commandline/latest/nuget.exe) (not [dotnet pack](dotnet-pack.md)).
```

Comment:
Good question. I just looked at the issues for Mono: https://github.com/NuGet/Home/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+mono

... so ... I guess as long as none of those would result in a big :boom: it should work. Before we state anything, let's ping @sayedihashimi to see if he knows.